### PR TITLE
Remove kSbUserPropertyHomeDirectory, make GetHomeDirectory w/o user

### PR DESCRIPTION
**Pull request by gbournou**
b/271930936

### CODE REVIEW

File: starboard/android/shared/get_home_directory.cc [hunk 1]
```diff
@@ -20,24 +20,19 @@
...
 #include "starboard/common/string.h"
-#include "starboard/shared/nouser/user_internal.h"
+#include "starboard/shared/starboard/get_home_directory.h"
 
 using ::starboard::android::shared::g_app_files_dir;
 
 namespace starboard {
 namespace shared {
-namespace nouser {
+namespace starboard {
```
Comment (kaidokert) [inline starboard/android/shared/get_home_directory.cc:29]: This seems weirdly out of place and a bit .. circular, but it matches the header, so makes sense. No action needed

**Review by kaidokert — APPROVED** [review]
Looks pretty clean - but please update the PR title and description to not have spill-over :) The longer description goes into the PR description and becomes the commit message body once merged.

File: starboard/shared/starboard/get_home_directory.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#ifndef STARBOARD_SHARED_STARBOARD_GET_HOME_DIRECTORY_H_
+#define STARBOARD_SHARED_STARBOARD_GET_HOME_DIRECTORY_H_
+
+#include "starboard/shared/internal_only.h"
+
+namespace starboard {
+namespace shared {
+namespace starboard {
+
+// A platform implementation of getting the home directory.
```
Comment (kaidokert) [inline starboard/shared/starboard/get_home_directory.h:24]: This wasn't clarified anywhere before, and a small nit, but perhaps add a short comment line to explain the implications of returning "false" from here.

Comment (kaidokert) [reply starboard/shared/starboard/get_home_directory.h:24]: Also looks like most implementations default to `kSbSystemPathCacheDirectory` which is maybe also worth calling out.

**Review by y4vor — APPROVED** [review]

**Pull request merged**